### PR TITLE
feat(create-signet): #MED-301 admin can create a signet

### PR DESCRIPTION
**Pull request by Mattox0**
## Describe your changes

A admin can open a modal and create a signet in admin signet page

## Checklist tests

- Access to admin page and try to create a signet

## Issue ticket number and link

[MED-301](https://jira.support-ent.fr/browse/MED-301)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/components/modals/create-signet/CreateSignet.tsx [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+        levels: levels
+          ?.filter((level: { label: string }) =>
+            selectedCheckboxes.levels.includes(level.label),
+          )
+          .map((level: { id: string; label: string }) => ({
+            id: level.id,
+            label: level.label,
+          })),
+        disciplines: disciplines
+          ?.filter((discipline: { label: string }) =>
```
Comment (ThomasSevagen) [inline frontend/src/components/modals/create-signet/CreateSignet.tsx:112]: .reduce

File: frontend/src/components/modals/create-signet/CreateSignet.tsx [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+  const handleDelete = (index: number, key: "levels" | "disciplines") => {
+    const newArray = selectedCheckboxes[key].filter((_, i) => i !== index);
+    setSelectedCheckboxesItems(key)(newArray);
+  };
+
+  const canCreateSignet = () => {
+    return (
+      title !== "" &&
+      url !== "" &&
+      selectedCheckboxes?.levels?.length > 0 &&
```
Comment (ThomasSevagen) [inline frontend/src/components/modals/create-signet/CreateSignet.tsx:169]: !!levels.lenfth

File: frontend/src/components/modals/create-signet/CreateSignet.tsx [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+      selectedCheckboxes?.disciplines?.length > 0 &&
```
Comment (ThomasSevagen) [inline frontend/src/components/modals/create-signet/CreateSignet.tsx:170]: same

File: frontend/src/hooks/useResourceListInfo.ts [hunk 4]
```diff
@@ -57,9 +61,9 @@ export const useResourceListInfo = (resources: Resource[] | null) => {
           acc.externalResources = [...acc.externalResources, resource];
           acc.types = [
             ...acc.types,
-            ...resource.document_types.filter(
-              (type) => !acc.types.includes(type),
-            ),
+            ...resource.document_types
```
Comment (ThomasSevagen) [inline frontend/src/hooks/useResourceListInfo.ts:64]: .reduce

File: frontend/src/routes/app/index.tsx [hunk 5]
```diff
@@ -280,9 +282,11 @@ export const App = () => {
           {alertText}
         </Alert>
       )}
-      <CreatePins refetch={refetchPins} />
-      <EditPins refetch={refetchPins} />
-      <ConfirmDelete refetch={refetchPins} />
+      {openModal === "create-pin" && <CreatePins refetch={refetchPins} />}
```
Comment (ThomasSevagen) [inline frontend/src/routes/app/index.tsx:285]: des CONST ou un ENUM ts?

File: frontend/src/components/modals/create-signet/CreateSignet.tsx [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+      console.error(error);
+    }
+  };
+
+  const uuidv = () => {
+    return "xxxxxxxx".replace(/[xy]/g, function (c) {
+      const r = (Math.random() * 16) | 0,
+        v = c == "x" ? r : (r & 0x3) | 0x8;
+      return v.toString(16);
+    });
```
Comment (niko-amosse) [inline frontend/src/components/modals/create-signet/CreateSignet.tsx:144]: il existe pas des fonctions qui font déjà ça ?

File: frontend/src/components/modals/create-signet/CreateSignet.tsx [hunk 7]
```diff
@@ -0,0 +1,355 @@
...
+      id="create-signet"
+    >
+      <Modal.Header onModalClose={handleCloseModal}>
+        {t("mediacentre.modal.signet.create.title")}
+      </Modal.Header>
+      <Modal.Subtitle>{t("mediacentre.modal.signet.subtitle")}</Modal.Subtitle>
+      <Modal.Body>
+        <Grid>
+          <Grid.Col
+            lg={width < 1280 ? "2" : "3"}
```
Comment (niko-amosse) [inline frontend/src/components/modals/create-signet/CreateSignet.tsx:210]: breakpoints

File: frontend/src/components/modals/create-signet/CreateSignet.tsx [hunk 8]
```diff
@@ -0,0 +1,355 @@
...
+              }}
+            />
+            {(thumbnail == "" || thumbnail == null) && thumbnailSrc == "" && (
+              <div className="font-red">
+                {t("mediacentre.modal.signet.ask.image")} *
+              </div>
+            )}
+          </Grid.Col>
+          <Grid.Col
+            lg={width < 1280 ? "6" : "9"}
```
Comment (niko-amosse) [inline frontend/src/components/modals/create-signet/CreateSignet.tsx:240]: idem

File: frontend/src/components/unique-image-picker/UniqueImagePicker.tsx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+const UniqueImagePicker: React.FC<UniqueImagePickerProps> = ({
+  onImageChange,
+  ...props
+}) => {
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const idRef = useRef(
+    `image-picker-${Math.random().toString(36).substring(2, 11)}`,
```
Comment (niko-amosse) [inline frontend/src/components/unique-image-picker/UniqueImagePicker.tsx:16]: idem ça n'existe pas ?

**Review by niko-amosse — APPROVED** [review]
amélioration possible : passer les nom des modales en constantes

File: frontend/src/components/modals/create-signet/CreateSignet.tsx (hunk 6 above)
Comment (Mattox0) [reply frontend/src/components/modals/create-signet/CreateSignet.tsx:144]: pas nativement, il faudrait installer une bibliothèque uniquement pour générer ce uuidv
(fonction récup du angular)

File: frontend/src/components/unique-image-picker/UniqueImagePicker.tsx (hunk 9 above)
Comment (Mattox0) [reply frontend/src/components/unique-image-picker/UniqueImagePicker.tsx:16]: idem que pour l'uuid (récupéré de magnéto)

**Review by niko-amosse — APPROVED** [review]

**Review by ThomasSevagen — APPROVED** [review]

**Pull request merged**